spec/services/update_course_stats_spec.rb

```diff
@@ -106,14 +106,14 @@
         subject
       end
       sentry_tag_uuid = subject.sentry_tag_uuid
-      expect(course.flags['update_logs'][1]['error_count']).to eq 8
+      expect(course.flags['update_logs'][1]['error_count']).to eq 328
```

Comment:
Before this PR, errors were logged at the batch level, resulting in a single Sentry log per batch, not per revision ID. For the specified course, the `update_revision_scores_for_course` method updates a total of 328 revisions (167 revisions + 161 parent revisions). Since all LiftWing requests are intentionally set to fail, there are 328 different Sentry logs.

Prior to these changes, there were only 8 logs because logs were at the batch level, and there were 8 batches: (161/50 + 1 = 4 batches) + (167/50 + 1 = 4 batches) = 8 batches.

I think that logging at revision level instead of batch level makes sense, because we're making one request to the liftwing API per revision id, so requests are completely individual. If we think that this could mean an overwhelming number of Sentry logs, I can make changes on the way errors are logged.

Reply:
It's a significant risk to go from batches to individual revisions for error logging. If there is a systematic problem where every request is failing, this could quickly saturate our Sentry event quota and result in further monitoring becoming unavailable.

It's good to include the counts in the Course update logs, as long as we're not actually sending that many separate Sentry events as well (if the same thing happened in production).

Reply:
Gotcha I rolled back some changes in order to keep logging errors at the batch level (50 revisions). Done mainly on c490de07f3fde4022a85e9e065b1ed629f2ee7c1. I think I should do something similar for the new `ReferenceCounterApi` class.

> It's good to include the counts in the Course update logs

That can be done, but we should implement some modifications. As I rolled back the changes, we're once again invoking `ApiErrorHandling.log_error` per batch, which, in turn, calls `UpdateServiceErrorHelper.update_error_stats` and increments `@error_count` by 1 each time. Despite passing the actual number of errors in the batch to `ApiErrorHandling.log_error`, we're not consistently utilizing that number to update `@error_count`, and it looks like that's the number we use for the 'error_count' course flag. I don't think that's in the scope of this PR, I could do it in a different one if we think it's important.

Reply:
It's not very important, but if you have a clear picture in your mind of what the problem is, please open an issue to document it.

Reply:
Created issue #5580